fastestimator/estimator/estimator.py

```diff
@@ -82,28 +82,30 @@ def _warmup(self):
             signature_epochs = list(set(epochs_pipeline) | set(epochs_network))
             state = {"mode": mode}
             for epoch in signature_epochs:
-                dataset = self.pipeline.dataset_schedule[mode].get_current_value(epoch)
-                batch = next(iter(dataset))
-                prediction = {}
-                batch = ChainMap(prediction, batch)
-                self.network.load_epoch(epoch, mode)
-                self.network.run_step(batch, state, warm_up=True)
+                ds_iter = self.pipeline.dataset_schedule[mode].get_current_value(epoch)
+                batch = next(ds_iter)
+                ops, model_list = self.network.load_epoch(epoch, mode)
+                model_list = self.network.model_schedule[mode].get_current_value(epoch)
+                self.network.run_step(batch, ops, model_list, state, warm_up=True)
 
     def train(self):
         self._run_traces_begin({"mode": "train"})
         train_step = 0
         for epoch in range(self.epochs):
-            dataset = self.pipeline.dataset_schedule["train"].get_current_value(epoch)
-            if self.steps_per_epoch:
-                dataset = dataset.take(self.steps_per_epoch)
+            ds_iter = self.pipeline.dataset_schedule["train"].get_current_value(epoch)
             batch_size = self.pipeline._get_batch_size(epoch)
-            self.network.load_epoch(epoch, "train")
+            if self.steps_per_epoch:
+                max_steps = self.steps_per_epoch
+            else:
+                max_steps = min(self.pipeline.num_examples["train"]) // batch_size
```

Comment:
if batch size is > number of samples we just do nothing right?

Reply:
yes.  nothing